src/domain/keywords/hooks/useKeywordsQueryStringWithReturnPath.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { addParamsToKeywordQueryString } from '../utils';
+
+const useKeywordsQueryStringWithReturnPath = (): string => {
+  const { pathname, search } = useLocation();
+  return React.useMemo(
+    () => addParamsToKeywordQueryString(search, { returnPath: pathname }),
```

Comment:
Have you measured the speed of this function? Not saying that this is redundant but maybe check if this is really causing noticeable perf issues.

Reply:
Did measure this, on my computer this `addParamsToKeywordQueryString(search, { returnPath: pathname })` takes about 0.01 - 0.02 milliseconds.

It is faster with useMemo but you have to decide between readability and small perf gain :)

Reply:
But I don't think this is something that necessarily needs to be changed.